src/data/list/basic.lean

```diff
@@ -684,6 +684,14 @@ by {induction l; intros, contradiction, reflexivity}
 by induction l;
   [refl, simp only [cons_append, last_cons (λ H, cons_ne_nil _ _ (append_eq_nil.1 H).2), *]]
 
+theorem last_append' (l₁ l₂ : list α) (h : l₂ ≠ []) :
```

Comment:
```suggestion
theorem last_append_of_ne_nil (l₁ l₂ : list α) (h : l₂ ≠ []) :
```

Reply:
Or maybe instead the lemma above should be renamed to `last_append_singleton`

Reply:
I agree that would be ideal, but I also don't want to create a backward compatibility problem if that is a concern.

Reply:
Mathlib doesn't care about backwards compatibility, but thanks for checking. Let's rename the existing lemma then instead, and remove the the `'` here?

Reply:
Yep, sounds good to me!

Reply:
Could you please implement the suggestion?

Reply:
The suggestion requires multiple changes in several files, so I am compiling mathlib to make sure that I don't screw anything up. It's been in the building state for 5 hours now.... I'll submit the change once it's done and everything type checks.

Reply:
I see, thanks!

Reply:
> I am compiling mathlib to make sure that I don't screw anything up.

Note that we have github set up to do this for you; and in my case, it's usually faster than building on my own machine. Rest assured if your code screws something up, CI will make sure we don't merge it till it compiles.

Reply:
Thanks -- done. I've also changed the label back to awaiting-review (hope I did it right).